### PR TITLE
Service Registration: Chunk

### PR DESCRIPTION
**Pull request by kmh287**
This pull request distinguishes installing/registering the chunk service and creating it.Please see #7839 for more context.

/to @choumx @jridgewell

### CODE REVIEW

File: src/chunk.js [hunk 1]
```diff
@@ -83,7 +83,23 @@ export function chunk(nodeOrAmpDoc, fn, priority) {
...
 export function chunkInstanceForTesting(nodeOrAmpDoc) {
-  return fromClassForDoc(nodeOrAmpDoc, 'chunk', Chunks);
+  return getChunkServiceForDoc_(nodeOrAmpDoc);
+}
+
+/**
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrAmpDoc
+ * @return {!Chunks}
+ */
+function getChunkServiceForDoc_(nodeOrAmpDoc) {
```
Comment (choumx) [inline src/chunk.js]: You can remove this method if you register the service in `startupChunk`. Registering and retrieving a service in the same method still looks weird to me.

File: src/chunk.js [hunk 2]
```diff
@@ -83,7 +83,23 @@ export function chunk(nodeOrAmpDoc, fn, priority) {
...
+  installChunkServiceForDoc_(nodeOrAmpDoc);
+  return getServiceForDoc(nodeOrAmpDoc, 'chunk');
+}
+
+/**
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrAmpDoc
+ */
+export function installChunkServiceForDoc_(nodeOrAmpDoc) {
```
Comment (choumx) [inline src/chunk.js]: Does this need to be exported? Note that trailing underscore is a naming convention for private props/functions.

If it's called in only one place, you probably don't need the method to start with.

Comment (kmh287) [inline src/chunk.js]: done

File: src/chunk.js (hunk 1 above)
Comment (kmh287) [inline src/chunk.js]: The function is called in three places. I'm not sure I see the value added by inlining it.

File: src/chunk.js [hunk 3]
```diff
@@ -107,7 +123,7 @@ export function activateChunkingForTesting() {
  * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrAmpDoc
  */
 export function runChunksForTesting(nodeOrAmpDoc) {
-  const service = fromClassForDoc(nodeOrAmpDoc, 'chunk', Chunks);
+  const service = getChunkServiceForDoc_(nodeOrAmpDoc);
```
Comment (kmh287) [inline src/chunk.js]: sure

File: src/chunk.js [hunk 4]
```diff
@@ -37,6 +37,15 @@ let deactivated = /nochunking=1/.test(self.location.hash);
 const resolved = Promise.resolve();
 
 /**
+ * @param {!Node|!./service/ampdoc-impl.AmpDoc} nodeOrAmpDoc
+ * @return {!Chunks}
```
Comment (kmh287) [inline src/chunk.js]: done

**Pull request merged**